Bottom: f93302c54a5796bde7c25ce29c6535ae4f11f78b
Top:    354f55f9e8af3af5a8e00c59008b53db06aa60aa
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2012-11-23 17:37:36 +0000

drc: HACK HACK HACK


---

diff --git a/src/find.c b/src/find.c
index 3f68165..5318a52 100644
--- a/src/find.c
+++ b/src/find.c
@@ -2610,7 +2610,12 @@ IsPolygonInPolygon (PolygonType *P1, PolygonType *P2)
 
               line.Point1.X = v->point[0];
               line.Point1.Y = v->point[1];
-              line.Thickness = 2 * Bloat;
+              line.Thickness = Bloat;
+              /* Another Bloat is added by IsLineInPolygon, making the correct
+               * 2x Bloat. Ideally we would change it there, but doing so
+               * breaks some other DRC checks which rely on the broken behaviour
+               * in IsLineInPolygon.
+               */
               line.Clearance = 0;
               line.Flags = NoFlags ();
               for (v = v->next; v != &c->head; v = v->next)
